sqlglot/dialects/duckdb.py

```diff
@@ -32,7 +32,7 @@ def _ts_or_ds_add(self, expression):
     this = self.sql(expression, "this")
     e = self.sql(expression, "expression")
     unit = self.sql(expression, "unit").strip("'") or "DAY"
-    return f"CAST({this} AS DATE) + INTERVAL {e} {unit}"
+    return f"CAST({this} AS DATE) + INTERVAL ({e}) {unit}"
```

Comment:
this is not really generic enough. can we refactor this in generator interval_sql?

1. use self.sql(exp.Interval)
2. in interval_sql in generator, only wrap in parens if needed, if it's not a literal
3. find all other occurrences of INTERVAL and refactor that to use this logic